### ISSUE TITLE
Making documentation of `compute_vertebral_distribution` more clear

### ISSUE THREAD

**Issue by NadiaBlostein** [spinalcordtoolbox/spinalcordtoolbox#4145]
# Overview

The following request pertains to the comment section of the `compute_vertebral_distribution` function defined in `spinalcordtoolbox/spinalcordtoolbox/types.py`. The original comment section (lines 333 to 342) goes as folllows:
```
"""
This function computes the vertebral distribution along the centerline, based on the position of
intervertebral discs in space. A reference label can be provided (default is top of C1) so that relative
distances are computed from this reference.

:param discs_levels: list: list of coordinates with value [[x, y, z, value], [x, y, z, value], ...]\
                the value correspond to the intervertebral disc label
:param label_reference: reference label from which relative position will be computed.\
                Must be on of self.labels_regions
"""
```

From my understanding, what coordinates you use for `:param discs_levels:` when you call `compute_vertebral_distribution` depends on the space where your Centerline object is defined. If you define the Centerline object in physical (`phys`) space, you need to call `compute_vertebral_distribution` with coordinates defined in physical space. If you define the Centerline object in pixel (`pix`) space, you need to call `compute_vertebral_distribution` with coordinates defined in pixel space. **See Steps 6 and 7 in the example below** to understand how I got to this conclusion. **See steps 1-5 in example below** to rerun it (apologies for how lengthy this is).

It took me a bit of time to figure this out; maybe specifying that the coordinates should be defined in the same space as the Centerline could save time for users in the future?

# Example
1. Download and unzip `data_batch-processing-of-subjects.zip` from [SCT tutorial](https://spinalcordtoolbox.com/user_section/tutorials/analysis-pipelines-with-sct/before-starting.html)

2. Install SCT & activate Miniconda environment (`Python 3.9.17`)
```
git clone https://github.com/spinalcordtoolbox/spinalcordtoolbox
cd spinalcordtoolbox
git checkout e740edf4c8408ffa44ef7ba23ad068c6d07e4b87
./install_sct
cd ..
conda activate spinalcordtoolbox/python/envs/venv_sct/
```

3. Current directory (`ls`):
- `multi_subject`
- `spinalcordtoolbox`

4. Label SC and discs according to [SCT tutorial](https://spinalcordtoolbox.com/user_section/tutorials/analysis-pipelines-with-sct/sct_run_batch-usage.html)
```
cd multi_subject
sct_run_batch -script process_data.sh -config config.yml
cd ..
```

5. Import correct libraries, load images and extract intervertebral disc coordinates (`Python 3.9.17`):
```
# modules and libraries
import os
import numpy as np
from spinalcordtoolbox.types import Centerline
from spinalcordtoolbox.centerline.core import ParamCenterline
from spinalcordtoolbox.centerline.core import get_centerline
from spinalcordtoolbox.image import Image

# defining paths and file names
current_path = os.getcwd()
path_data = current_path + '/multi_subject/data/'
path_output = current_path + '/multi_subject/output/data_processed/'
subject_name = 'sub-01'
fname_image_seg = path_output + subject_name +  '/anat/' + subject_name + '_T2w_seg.nii.gz'
fname_image_discs = path_output + subject_name +  '/anat/' + subject_name + '_T2w_seg_labeled_discs.nii.gz'

# loading images
im_seg = Image(fname_image_seg).change_orientation('RPI')
im_discs = Image(fname_image_discs).change_orientation('RPI')

# extracting intervertebral discs
coord = im_discs.getNonZeroCoordinates(sorting = 'z', reverse_coord = True)
coord_pix = []
coord_phys = []
for c in coord:
    coord_pix.append(list(c))
    c_p = list(im_discs.transfo_pix2phys([[c.x, c.y, c.z]])[0])
    c_p.append(c.value)
    coord_phys.append(c_p)

# Defining centerline parameters
param_centerline = ParamCenterline(algo_fitting = 'linear', smooth = 50, degree = None, minmax = None)
```
6. **Test 1:** Extracting Centerline in physical space
```
_, arr_ctl_phys, arr_ctl_der_phys, _ = get_centerline(im_seg, param = param_centerline = , space = 'phys')
centerline_phys = Centerline(points_x = arr_ctl_phys[0], points_y = arr_ctl_phys[1], points_z = arr_ctl_phys[2], 
    deriv_x = arr_ctl_der_phys[0], deriv_y = arr_ctl_der_phys[1], deriv_z = arr_ctl_der_phys[2])

# CORRECT: compute vertebral distribution using physical coordinates
centerline_phys.compute_vertebral_distribution(coord_phys)
print(centerline.distance_from_C1label)

# INCORRECT: compute vertebral distribution using pixel coordinates
centerline_phys.compute_vertebral_distribution(coord_phys)
print(centerline.distance_from_C1label)
```
7. **Test 2:** Extracting Centerline in pixel space
```
_, arr_ctl_pix, arr_ctl_der_pix, _ = get_centerline(im_seg, param = param_centerline = , space = 'pix')
centerline_pix = Centerline(points_x = arr_ctl_pix[0], points_y = arr_ctl_pix[1], points_z = arr_ctl_pix[2], 
    deriv_x = arr_ctl_der_pix[0], deriv_y = arr_ctl_der_pix[1], deriv_z = arr_ctl_der_pix[2])

# CORRECT: compute vertebral distribution using pixel coordinates
centerline_pix.compute_vertebral_distribution(coord_pix)
print(centerline.distance_from_C1label)

# INCORRECT: compute vertebral distribution using physical coordinates
centerline_pix.compute_vertebral_distribution(coord_phys)
print(centerline.distance_from_C1label)
```

# Moving forward
Previously (`spinalcordtoolbox/spinalcordtoolbox/types.py`, lines `338-339`): 
```
        :param discs_levels: list: list of coordinates with value [[x, y, z, value], [x, y, z, value], ...]\
                        the value correspond to the intervertebral disc label
```
Proposed new explanation
```
        :param discs_levels: list: list of coordinates with value [[x, y, z, value], [x, y, z, value], ...]\
                        the value corresponds to the intervertebral disc label. These coordinates should\
                        be defined in the same space as the Centerline (`self`) object. This means that if\
                        your Centerline object is defined in physical (`phys`) space, the [x,y,z] coordinates\
                        should be defined in physical space. If your Centerline object is defined in pixel \
                        (`pix`) space, the [x,y,z] coordinates should be defined in pixel space.         
```

**Comment by joshuacwnewton** [comment]
I absolutely agree! Feel free to open a PR by editing the function's docstring in the `types.py` file.

For such a small documentation change, IMO it's easiest to do this directly on GitHub rather than bothering with a fork or anything like that -- you should be able to head to [`types.py`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/master/spinalcordtoolbox/types.py) and hit the :pen: edit button, make your changes, commit them, and then GitHub will open up a branch and PR for you. For example:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/a3c45db5-5b10-4d26-ad4a-1fd1f402258e)

**Comment by NadiaBlostein** [comment]
# TLDR
From my understanding, what coordinates you use for `:param discs_levels:` when you call `compute_vertebral_distribution` depends on the space where your Centerline object is defined. If you define the Centerline object in physical (`phys`) space, you need to call `compute_vertebral_distribution` with coordinates defined in physical space. If you define the Centerline object in pixel (`pix`) space, you need to call `compute_vertebral_distribution` with coordinates defined in pixel space. See Steps 6 and 7 in example below (steps 1-5 are just there if you want to rerun it, apologies for how long this looks).

# Example
1. Download and unzip `data_batch-processing-of-subjects.zip` from [SCT tutorial](https://spinalcordtoolbox.com/user_section/tutorials/analysis-pipelines-with-sct/before-starting.html)

2. Install SCT & activate Miniconda environment (`Python 3.9.17`)
```
git clone https://github.com/spinalcordtoolbox/spinalcordtoolbox
cd spinalcordtoolbox
git checkout e740edf4c8408ffa44ef7ba23ad068c6d07e4b87
./install_sct
cd ..
conda activate spinalcordtoolbox/python/envs/venv_sct/
```

3. Current directory (`ls`):
- `multi_subject`
- `spinalcordtoolbox`

4. Label SC and discs according to [SCT tutorial](https://spinalcordtoolbox.com/user_section/tutorials/analysis-pipelines-with-sct/sct_run_batch-usage.html)
```
cd multi_subject
sct_run_batch -script process_data.sh -config config.yml
cd ..
```

5. Import correct libraries, load images and extract intervertebral disc coordinates (`Python 3.9.17`):
```
# modules and libraries
import os
import numpy as np
from spinalcordtoolbox.types import Centerline
from spinalcordtoolbox.centerline.core import ParamCenterline
from spinalcordtoolbox.centerline.core import get_centerline
from spinalcordtoolbox.image import Image

# defining paths and file names
current_path = os.getcwd()
path_data = current_path + '/multi_subject/data/'
path_output = current_path + '/multi_subject/output/data_processed/'
subject_name = 'sub-01'
fname_image_seg = path_output + subject_name +  '/anat/' + subject_name + '_T2w_seg.nii.gz'
fname_image_discs = path_output + subject_name +  '/anat/' + subject_name + '_T2w_seg_labeled_discs.nii.gz'

# loading images
im_seg = Image(fname_image_seg).change_orientation('RPI')
im_discs = Image(fname_image_discs).change_orientation('RPI')

# extracting intervertebral discs
coord = im_discs.getNonZeroCoordinates(sorting = 'z', reverse_coord = True)
coord_pix = []
coord_phys = []
for c in coord:
    coord_pix.append(list(c))
    c_p = list(im_discs.transfo_pix2phys([[c.x, c.y, c.z]])[0])
    c_p.append(c.value)
    coord_phys.append(c_p)

# Defining centerline parameters
param_centerline = ParamCenterline(algo_fitting = 'linear', smooth = 50, degree = None, minmax = None)
```
6. **Test 1:** Extracting Centerline in physical space
```
_, arr_ctl_phys, arr_ctl_der_phys, _ = get_centerline(im_seg, param = param_centerline = , space = 'phys')
centerline_phys = Centerline(points_x = arr_ctl_phys[0], points_y = arr_ctl_phys[1], points_z = arr_ctl_phys[2], 
    deriv_x = arr_ctl_der_phys[0], deriv_y = arr_ctl_der_phys[1], deriv_z = arr_ctl_der_phys[2])

# CORRECT: compute vertebral distribution using physical coordinates
centerline_phys.compute_vertebral_distribution(coord_phys)
print(centerline.distance_from_C1label)

# INCORRECT: compute vertebral distribution using pixel coordinates
centerline_phys.compute_vertebral_distribution(coord_phys)
print(centerline.distance_from_C1label)
```
7. **Test 2:** Extracting Centerline in pixel space
```
_, arr_ctl_pix, arr_ctl_der_pix, _ = get_centerline(im_seg, param = param_centerline = , space = 'pix')
centerline_pix = Centerline(points_x = arr_ctl_pix[0], points_y = arr_ctl_pix[1], points_z = arr_ctl_pix[2], 
    deriv_x = arr_ctl_der_pix[0], deriv_y = arr_ctl_der_pix[1], deriv_z = arr_ctl_der_pix[2])

# CORRECT: compute vertebral distribution using pixel coordinates
centerline_pix.compute_vertebral_distribution(coord_pix)
print(centerline.distance_from_C1label)

# INCORRECT: compute vertebral distribution using physical coordinates
centerline_pix.compute_vertebral_distribution(coord_phys)
print(centerline.distance_from_C1label)
```

# Moving forward
Previously (`spinalcordtoolbox/spinalcordtoolbox/types.py`, lines `338-339`): 
```
        :param discs_levels: list: list of coordinates with value [[x, y, z, value], [x, y, z, value], ...]\
                        the value correspond to the intervertebral disc label
```
Proposed new explanation
```
        :param discs_levels: list: list of coordinates with value [[x, y, z, value], [x, y, z, value], ...]\
                        the value corresponds to the intervertebral disc label. These coordinates should\
                        be defined in the same space as the Centerline (`self`) object. This means that if\
                        your Centerline object is defined in physical (`phys`) space, the [x,y,z] coordinates\
                        should be defined in physical space. If your Centerline object is defined in pixel \
                        (`pix`) space, the [x,y,z] coordinates should be defined in pixel space.         
```

**Comment by mguaypaq** [comment]
Looking at the code for the `Centerline` object, I agree with the results of your investigation: the input data is a bunch of coordinates gotten from `get_centerline()`, and no image data or metadata (that could be used to convert between physical and pixel space). So, the methods of `Centerline` pretty much have to be in the same space as the input data.

I'll put your suggested changes in a PR.
